test/e2e/run/e2e.go

```diff
@@ -0,0 +1,623 @@
+	"io"
+	"io/fs"
+	"os"
+	"os/exec"
+	"os/signal"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/go-logr/logr"
+	ginkgoTypes "github.com/onsi/ginkgo/v2/types"
+	"gopkg.in/yaml.v2"
+
+	"github.com/aws/eks-hybrid/test/e2e"
+	"github.com/aws/eks-hybrid/test/e2e/cleanup"
+	"github.com/aws/eks-hybrid/test/e2e/cluster"
+	"github.com/aws/eks-hybrid/test/e2e/constants"
+	"github.com/aws/eks-hybrid/test/e2e/suite"
+)
+
+const (
+	e2eConfigFolder      = "configs"
+	e2eConfigFile        = "e2e-param.yaml"
+	e2eTestResourcesFile = "e2e-test-resources.yaml"
+	e2eReportsFolder     = "reports"
+	e2eReportsFile       = "junit-nodeadm.xml"
+	e2eReportsFileJSON   = "junit-nodeadm.json"
+	serialOutputLogFile  = "serial-output.log"
+	testCleanupLogFile   = "cleanup-output.log"
+	testGinkgoOutputLog  = "ginkgo-output.log"
+	testSetupLogFile     = "setup-output.log"
+)
+
+type E2E struct {
+	AwsCfg            aws.Config
+	CurrentPhase      string
+	Logger            logr.Logger
+	NoColor           bool
+	TestConfig        suite.TestConfig
+	TestLabelFilter   string
+	TestsFailed       bool
+	TestProcs         int
+	TestTimeout       string
+	TestsBinaryOrPath string
+	TestResources     cluster.TestResources
+	SkipCleanup       bool
+	SkippedTests      string
+}
+
+type E2EResult struct {
+	ArtifactsBucketPath string       `json:"artifacts_bucket_path"`
+	CleanupLog          string       `json:"cleanup_log"`
+	FailedTests         []FailedTest `json:"failed_tests"`
+	GinkgoLog           string       `json:"ginkgo_log"`
+	LastPhase           string       `json:"last_phase"`
+	RunFailureMessage   string       `json:"run_failure_message"`
+	SetupLog            string       `json:"setup_log"`
+	TestRan             int          `json:"test_ran"`
+	TotalTests          int          `json:"total_tests"`
+}
+
+type FailedTest struct {
+	ArtifactsBucketPath string `json:"artifacts_bucket_path"`
+	CollectorLogsBundle string `json:"collector_logs_bundle"`
+	FailureMessage      string `json:"failure_message"`
+	InstanceName        string `json:"instance_name"`
+	GinkgoLog           string `json:"ginkgo_log"`
+	Name                string `json:"name"`
+	SerialLog           string `json:"serial_output_log"`
+	State               string `json:"state"`
+}
+
+func (e *E2E) Run(ctx context.Context) error {
+	// continue even on error to output as much as possible
+	err := e.runTests(ctx)
+
+	e.updateE2EResultPhase("parse-report")
+	e2eResult, reportErr := e.parseReport()
+	if reportErr != nil {
+		err = errors.Join(err, fmt.Errorf("parsing report: %w", reportErr))
+	}
+
+	e.updateE2EResultPhase("upload-artifacts-to-s3")
+	s3Success := true
+	s3Err := e.uploadArtifactsToS3(ctx)
+	if s3Err != nil {
+		err = errors.Join(err, fmt.Errorf("uploading logs to s3: %w", s3Err))
+		s3Success = false
+	}
+
+	e2eResult.LastPhase = e.CurrentPhase
+	e2eResult = e.updateE2EResultLogs(e2eResult, s3Success)
```

Comment:
i got lost here, sorry
What is this for and why do we need to update the paths in the test results? why can't we generate the right path from the beginning?